relay/client/manager.go

```diff
@@ -86,29 +88,66 @@ func (m *Manager) Serve() error {
 	if m.relayClient != nil {
 		return fmt.Errorf("manager already serving")
 	}
+	log.Debugf("starting relay client manager with %v relay servers", m.serverURLs)
 
-	m.relayClient = NewClient(m.ctx, m.serverURL, m.tokenStore, m.peerID)
-	err := m.relayClient.Connect()
-	if err != nil {
-		log.Errorf("failed to connect to relay server: %s", err)
-		return err
+	successChan := make(chan *Client, 1)
+
+	ctx, cancel := context.WithTimeout(m.ctx, connectionTimeout)
+	defer cancel()
+
+	sem := make(chan struct{}, maxConcurrentServers)
+
+	for _, url := range m.serverURLs {
+		sem <- struct{}{}
+		go func(url string) {
+			defer func() { <-sem }()
+			m.connect(m.ctx, url, successChan)
+		}(url)
 	}
 
-	m.reconnectGuard = NewGuard(m.ctx, m.relayClient)
-	m.relayClient.SetOnDisconnectListener(func() {
-		m.onServerDisconnected(m.serverURL)
-	})
-	m.startCleanupLoop()
+	// Wait for the first successful connection or all attempts to fail
+	select {
```

Comment:
What is the goal of this logic?
How will look the server, and load balancer infrastructure? This parallel connection attempt approach will not be ended on the same server with two different URLs? The original idea was that every peer tries to connect to the same URL and the server-side load balancer manages the server instance assigmenet.